avocado/utils/configure_network.py

```diff
@@ -92,3 +92,16 @@ def unset_ip(interface):
         return True
     except Exception as ex:
         raise NWException("ifdown fails: %s" % ex)
+
+
+def interfacewait(interface):
+    """
+    Waits for the interface link to be UP
+    """
+    for _ in range(0, 600, 5):
+        if 'UP' or 'yes' in \
+           process.system_output("timeout 120 ip link show %s | head -1"
```

Comment:
Please avoid using binaries / shell functionality in this *Python* code base.  On this one line you require:
* timeout: there's `avocado.utils.wait`
* ip: please consider checking if either `/proc` os `/sys` expose the information you're looking for
* head: consider getting the first line of a file/output the Python way